### ISSUE TITLE
Only generating four bounding boxes per aspect ratio?

### ISSUE THREAD

**Issue by LaurensHagendoorn** [fizyr/keras-retinanet#161]
Could there be any reason why the network is limited in the number of bounding boxes it finds per aspect ratio? I have images containing 6 annotations roughly equal in size and aspect ratios. The data set should not be too hard either. However, the network only ever detects four of the objects.

**Comment by hgaiser** [comment]
There probably is a reason, but based on the information given it's difficult to guess what it is :p 

What do you mean with "finds per aspect ratio"?

**Comment by LaurensHagendoorn** [comment]
Well, even when training on the validation set (thus major overfitting) and visualising detections on this set, it only finds 4 correct bounding boxes of the right aspect ratio, and seemingly one of a different shape. (at a score treshold of 0.3)

![102133939_0_11_vn2_ms6552_display_analog](https://user-images.githubusercontent.com/32707472/33654687-27bda7fc-da71-11e7-91d3-32195460a82e.png)

**Comment by hgaiser** [comment]
Are you using the same preprocessing as during training? In other words, are you using the CSV generator to feed this image or are you loading your own image and feeding that to the network?

**Comment by LaurensHagendoorn** [comment]
I am using the same generator (I slightly adapted the pascal generator, but it's pretty much that)

**Comment by hgaiser** [comment]
Then I don't have any other tips for you I'm afraid.

**Comment by LaurensHagendoorn** [comment]
Alright, then I will just try some more things I guess, super odd behaviour. I suppose I could choose more similar aspect ratio's and scales for the anchors.

**Comment by hgaiser** [comment]
As a sanity check you could pad your images such that they are square (say `512x512`).

**Comment by LaurensHagendoorn** [comment]
Could the shape of the image be the issue? I suppose it is a bit non-standard, I'll try that!

**Comment by hgaiser** [comment]
It shouldn't, but as you said it's non-standard so worth a shot.

**Comment by hgaiser** [comment]
I'll close this issue for now then, let me know how it turns out.
